### PR TITLE
Devroot cleanup

### PR DESCRIPTION
**Pull request by devinrsmith**
This removes the necessity for the `devroot` property to be set - it doesn't make sense for end-users. In code-generation or test cases where access to the source is still necessary, the appropriate system properties can be set as necessary.

### CODE REVIEW

File: Generators/src/main/java/io/deephaven/libs/GroovyStaticImportGenerator.java [hunk 1]
```diff
@@ -373,15 +372,13 @@ public static void main(String[] args) throws ClassNotFoundException, IOExceptio
...
-            devroot = Configuration.getInstance().getDevRootPath();
-        } else if (args.length == 1) {
+        if (args.length == 1) {
             devroot = args[0];
         } else if (args.length == 2) {
             devroot = args[0];
             assertNoChange = Boolean.parseBoolean(args[1]);
         } else {
-            System.out.println("Usage: [<devroot> [assertNoChange]]");
+            System.out.println("Usage: <devroot> [assertNoChange]");
```
Comment (rcaudy) [inline Generators/src/main/java/io/deephaven/libs/GroovyStaticImportGenerator.java:381]: Have you tested all of the generators?

File: Integrations/src/main/java/io/deephaven/integrations/python/PythonDeephavenSession.java [hunk 2]
```diff
@@ -56,9 +56,7 @@ public class PythonDeephavenSession extends AbstractScriptSession<PythonSnapshot
     private static final Logger log = LoggerFactory.getLogger(PythonDeephavenSession.class);
 
     private static final String DEFAULT_SCRIPT_PATH = Configuration.getInstance()
-            .getProperty("PythonDeephavenSession.defaultScriptPath")
-            .replace("<devroot>", Configuration.getInstance().getDevRootPath())
-            .replace("<workspace>", Configuration.getInstance().getWorkspacePath());
+            .getStringWithDefault("PythonDeephavenSession.defaultScriptPath", ".");
```
Comment (rcaudy) [inline Integrations/src/main/java/io/deephaven/integrations/python/PythonDeephavenSession.java:59]: Are we confident about making `.` the new default for all the script sessions? What testing have you done?

Comment (rcaudy) [inline Integrations/src/main/java/io/deephaven/integrations/python/PythonDeephavenSession.java:59]: Related to this change:
`io.deephaven.engine.util.ScriptFinder#findScriptEx(java.lang.String, java.lang.String)` has some comments that probably need cleanup after devroot is retired.

Comment (devinrsmith) [reply Integrations/src/main/java/io/deephaven/integrations/python/PythonDeephavenSession.java:59]: This previous defaults were `<X>.defaultScriptPath=<devroot>`; and by default, `devroot=.`. So in theory, it's the same defaults as before, just a layer of indirection removed.

I think the defaultScriptPath logic could use serious reworking - I'm not sure it makes sense anymore if we have a better way w/ application mode. There's a comment to this effect in `io.deephaven.engine.util.ScriptFinder#findScriptEx`

I've run locally natively in python and groovy. I'll do testing in docker as well. I'll start a nightly as well.

Comment (devinrsmith) [reply Integrations/src/main/java/io/deephaven/integrations/python/PythonDeephavenSession.java:59]: Ha! Just mentioned this in the previous comment before I noticed this.

File: Generators/src/main/java/io/deephaven/libs/GroovyStaticImportGenerator.java (hunk 1 above)
Comment (devinrsmith) [reply Generators/src/main/java/io/deephaven/libs/GroovyStaticImportGenerator.java:381]: I've run `./gradlew Generators:generateAll` and I think testing should also assert all the generators are correct.

File: Integrations/src/main/java/io/deephaven/integrations/python/PythonDeephavenSession.java (hunk 2 above)
Comment (devinrsmith) [reply Integrations/src/main/java/io/deephaven/integrations/python/PythonDeephavenSession.java:59]: Everything I've run seems to work. nightly is passing except I hit this https://github.com/deephaven/deephaven-core/issues/2732

Comment (devinrsmith) [reply Integrations/src/main/java/io/deephaven/integrations/python/PythonDeephavenSession.java:59]: Also run python embedded server, work fine.

Comment (devinrsmith) [reply Integrations/src/main/java/io/deephaven/integrations/python/PythonDeephavenSession.java:59]: https://github.com/deephaven/deephaven-core/issues/2978

**Review by rcaudy — APPROVED** [review]

**Review by jcferretti — APPROVED** [review]

**Pull request merged**